### ISSUE TITLE
OEL9: Cannot connect to legacy systems, getting error message "no matching host key type found. Their offer: ssh-rsa" 

### ISSUE THREAD

**Issue by ken-bell** [oracle/oracle-linux#160]
Trying to connect from an OEL9 system in FIPS mode to a legacy system (CentOS6), the following error message is displayed
    [Raw](https://access.redhat.com/solutions/7056072#)

$ ssh legacy-system
    Unable to negotiate with IPAddr port 22: no matching host key type found. Their offer: ssh-rsa

Trying to connect from an OEL8 system works fine

This was described and apparently fixed in https://access.redhat.com/solutions/7056072

**Comment by YoderExMachina** [comment]
Oracle Linux customers, please file your issue at https://support.oracle.com

Thanks for filing an issue with Oracle Linux. 

GitHub Issues is not an official support channel and we don't offer 
product support here. If you're not yet an Oracle Linux customer, 
consider signing up at https://linux.oracle.com. 

Even if you're not a customer, if we can confirm that an issue is a 
bug we will do our best to fix it and to update this issue 
once it has been fixed. We don't guarantee a fix or feedback and
for now, we will close this issue. If you have Oracle Linux support, 
please use support.oracle.com to report issues.

**Comment by hbergh** [comment]
OL9 is compatible with RHEL9 so the solution offered above should work the same on OL9. If not, please let us know what part of that solution is working differently in OL9 compared to RHEL9.

**Comment by ken-bell** [comment]
Thank you for replying, hbergh!

I tried RHEL's "Solution 1 - regenerate keys on the legacy system", via generating an ECDSA key on the server, adding it to the sshd_config there, and restarting sshd.  I assume (don't have a RHEL9 system to test with) this works on RHEL9 given that they present it as a solution, but it didn't work on OEL9 (client) with CentOS6 (server).  The OEL9 client runs "OpenSSH_8.7p1, OpenSSL 3.0.7 1 Nov 2022" and the CentOS6 server runs "OpenSSH_5.3p1, OpenSSL 1.0.1e-fips 11 Feb 2013".  Thank you!

**Comment by hbergh** [comment]
Did you also generate an ecdsa key pair and add the id_ecdsa.pub to .ssh/authorized_keys of the Centos6 user? What was the exact error message you got, and can you provide `ssh -vvv` output?

**Comment by ken-bell** [comment]
No, I didn't generate an ECDSA key pair on the OEL9 client; the error message complained about a "host key" so I didn't think that would be relevant. Am I wrong on that?  Normally I expect that when a server's host key changes, the SSH client prompts for whether to accept the new key.

Given that the server apparently continued to offer only "ssh-rsa", I took that to mean that it wasn't presenting the ECDSA host key, and assumed then that the server didn't handle ECDSA host keys.  I didn't think that the issue was the client's public key pair, I didn't think that should have required any change.  Again, maybe I was wrong?

The error message was the same as in the RHEL "solution":

```
(OEL9) $ ssh legacy-system
Unable to negotiate with IPAddr port 22: no matching host key type found. Their offer: ssh-rsa
```
Unfortunately I didn't save the verbose output (it all looked normal except for that error message), and I reverted the system (removed the ECDSA key) because it's a production server and that caused other problems for a user.  If need be I can probably redo the change to capture the "-vvv" output.

Thank you -- Ken
